### PR TITLE
Create e2e-test-writing-guide-cn.md

### PR DESCRIPTION
**Pull request by likyh**
add 「How to Add Some E2E Test for Plugin?」

### CODE REVIEW

File: docs/09-DeveloperDoc/e2e-test-writing-guide-cn.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
+# 如何为插件编写E2E测试
+
+## 为什么要写 E2E 测试
+
+E2E 测试，作为自动化测试的一环，一般是指白盒的集成测试，或者允许使用一些数据库等外部服务的单元测试。书写E2E测试的目的是屏蔽一些内部实现逻辑，仅从数据正确性的角度来看同样的外部输入，是否可以得到同样的输出。另外，相较于黑盒的集成测试来说，可以避免一些网络等因素带来的偶然问题。更多关于插件的介绍，可以在这里获取更多信息： 为什么要编写 E2E 测试（未完成）
```
Comment (klesh) [inline docs/09-DeveloperDoc/e2e-test-writing-guide-cn.md]: I think it is "黑", not "白", as you say later on "屏蔽一些内部实现逻辑"

File: docs/09-DeveloperDoc/e2e-test-writing-guide-cn.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+	// import raw data table
+	dataflowTester.ImportCsvIntoRawTable("./raw_tables/_raw_feishu_meeting_top_user_item.csv", "_raw_feishu_meeting_top_user_item")
+
+	// verify extraction
+	dataflowTester.FlushTabler(&models.FeishuMeetingTopUserItem{})
+	dataflowTester.Subtask(tasks.ExtractMeetingTopUserItemMeta, taskData)
+
+}
+```
+新增的代码包括调用`dataflowTester.FlushTabler`刷新会议表，调用`dataflowTester.Subtask`模拟子任务`ExtractMeetingTopUserItemMeta`的运行。
```
Comment (klesh) [inline docs/09-DeveloperDoc/e2e-test-writing-guide-cn.md]: 刷新什么表？

Comment (likyh) [reply docs/09-DeveloperDoc/e2e-test-writing-guide-cn.md]: fixed

File: i18n/zh/docusaurus-plugin-content-docs/current/09-DeveloperDoc/e2e-test-writing-guide.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+}
+```
+它的功能是调用`dataflowTester.VerifyTable`完成数据结果的验证。第三个参数是表的主键，第四个参数是表所有需要验证的字段。用于验证的数据存在`./snapshot_tables/_tool_feishu_meeting_top_user_items.csv`中，当然，目前此文件还不存在。
+
+为了方便生成前述文件，DevLake采取了一种称为`Snapshot`的测试技术，在调用`VerifyTable`文件且csv不存在时，将会根据运行结果自动生成文件。
+
+但注意！自动生成后并不是高枕无忧，还需要做两件事：1. 检查文件生成是否正确 2. 再次运行，以便于确定生成结果和再次运行的结果没有差错。
+这两项操作非常重要，直接关系到测试编写的质量，我们应该像对待代码文件一样对待`.csv`格式的 snapshot 文件。
+
+如果这一步出现了问题，一般会是2种问题，
```
Comment (klesh) [inline i18n/zh/docusaurus-plugin-content-docs/current/09-DeveloperDoc/e2e-test-writing-guide.md:177]: 一般会有2种可能

File: i18n/zh/docusaurus-plugin-content-docs/current/09-DeveloperDoc/e2e-test-writing-guide.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+1. 验证的字段中含有类似create_at运行时间或者自增id的字段，这些无法重复验证的字段应该排除。
+2. 运行的结果中存在`\n`或`\r\n`等转义不匹配的字段，一般是解析`httpResponse`时出现的错误，可以参考如下方案解决：
+    1. 修改api模型中，内容的字段类型为`json.RawMessage`
+    2. 在解析时再将其转化为string
+    3. 如此操作，即可原封不动的保存`\n`符号，避免数据库或操作系统对换行符的解析
```
Comment (klesh) [inline i18n/zh/docusaurus-plugin-content-docs/current/09-DeveloperDoc/e2e-test-writing-guide.md:182]: 1 跟 2 是前后关系的话，第3点开头用“经过以上操作，”为开头感觉会更清晰

Comment (likyh) [reply i18n/zh/docusaurus-plugin-content-docs/current/09-DeveloperDoc/e2e-test-writing-guide.md:182]: fixed

**Comment by likyh** [comment]
reopen in https://github.com/apache/incubator-devlake-website/pull/105

**Pull request closed without merging**